nextcord/ui/select/channel.py

```diff
@@ -0,0 +1,254 @@
+            custom_id=custom_id,
+            type=ComponentType.channel_select,
+            placeholder=placeholder,
+            min_values=min_values,
+            max_values=max_values,
+            disabled=disabled,
+            **kwargs,
+        )
+        self.row = row
+
+    @property
+    def options(self) -> List[SelectOption]:
+        """List[:class:`nextcord.SelectOption`]: A list of options that can be selected in this menu.
+        This will always be an empty list since channel selects cannot have any options.
+        """
+        return []
+
+    @property
+    def values(self) -> List[int]:
+        """List[:class:`int`]: A list of channel ids that have been selected by the user."""
+        return [int(id) for id in self._selected_values]
+
+    def get_channels(self, guild: Guild) -> List[Union[GuildChannel, Thread]]:
+        """A shortcut for getting all :class:`nextcord.abc.GuildChannel`'s of :attr:`.values`.
+        Channels that are not found in cache will not be returned.
+        To get all channels regardless of whether they are in cache or not, use :meth:`.fetch_channels`.
+
+        Parameters
+        ----------
+        guild: :class:`nextcord.Guild`
+            The guild to get the channels from.
+
+        Returns
+        -------
+        List[Union[:class:`nextcord.abc.GuildChannel`, :class:`nextcord.Thread`]]
+            A list of channels that have been selected by the user.
+        """
+        channels: List[Union[GuildChannel, Thread]] = []
+        for id in self.values:
+            channel = guild.get_channel_or_thread(id)
+            if channel is not None:
+                channels.append(channel)
+        return channels
+
+    async def fetch_channels(self, guild: Guild) -> List[Union[GuildChannel, Thread]]:
+        """A shortcut for fetching all :class:`nextcord.abc.GuildChannel`'s of :attr:`.values`.
+        Channels that are not found in cache will be fetched.
+
+        Parameters
+        ----------
+        guild: :class:`nextcord.Guild`
+            The guild to get the channels from.
+
+        Raises
+        ------
+        :exc:`.HTTPException`
+            Fetching the channels failed.
+        :exc:`.NotFound`
+            A channel was not found.
+        :exc:`.Forbidden`
+            You do not have the proper permissions to fetch a channel.
+        :exc:`.InvalidArgument`
+            The channel type is not supported.
+
+        Returns
+        -------
+        List[Union[:class:`nextcord.abc.GuildChannel`, :class:`nextcord.Thread`]]
+            A list of channels that have been selected by the user.
+        """
+        channels: List[Union[GuildChannel, Thread]] = []
+        guild_channels = None
+        for id in self.values:
+            channel = guild.get_channel_or_thread(id)
+            if channel is None:
+                if guild_channels is None:
+                    guild_channels = await guild.fetch_channels()
```

Comment:
I'm confused on why you can't just fetch the channels before the first for loop runs?

Reply:
Because if every channel is cached, it does not need to fetch them again

Reply:
But the code before the for loop isn't going to run multiple times?

Reply:
Ohhhh wait nevermind I get it now